### PR TITLE
Adapt to new slackclient interface

### PR DESCRIPTION
**Pull request by peterdemin**
In [PR-54](https://github.com/slackhq/python-slackclient/pull/54) slackclient is changing return type of api_call, simplifying it's usage.

### CODE REVIEW

File: errbot/backends/slack.py [hunk 1]
```diff
@@ -200,7 +199,7 @@ def api_call(self, method, data=None, raise_errors=True):
         """
         if data is None:
             data = {}
-        response = json.loads(self.sc.server.api_call(method, **data).decode('utf-8'))
+        response = self.sc.server.api_call(method, **data).json()
```
Comment (zoni) [inline errbot/backends/slack.py]: In the referenced PR, [line 135](https://github.com/slackhq/python-slackclient/pull/54/files#diff-599dbee7d2139b2e66b89a120d50a0a1R135) returns `Response.text`, which returns a unicode object. This means it should still be decoded as `utf-8` and fed to `json.loads` (there's no `.json()` available, that's only a function on the original `Response` object).

That means there should be no changes necessary on our side, which in my opinion is how it should be. You don't want to change the public API, as that might break a lot of users of library.

**Comment by gbin** [comment]
It doesn't look upstream has been merged, what is the status about this ?

**Comment by peterdemin** [comment]
There was no action in python-slackclient since Nov 25.
However in response to zoni's comment I think, I should change PR-54 to keep public API the same.

**Comment by zoni** [comment]
> However in response to zoni's comment I think, I should change PR-54 to keep public API the same.

Looking at the diff of https://github.com/slackhq/python-slackclient/pull/54, nothing about the public API changes though.

**Comment by peterdemin** [comment]
`SlackRequest.do` was returning `urlopen()` result, and I changed it to `requests.post()` result that is not compatible.

**Comment by gbin** [comment]
yes it brakes:
   response = json.loads(self.sc.server.api_call(method, **data).decode('utf-8'))
AttributeError: 'str' object has no attribute 'decode'

**Comment by gbin** [comment]
but yours too -_-
    response = self.sc.server.api_call(method, **data).json()
AttributeError: 'str' object has no attribute 'json'

**Comment by gbin** [comment]
Pushed the fix directly on master but it still completely fails for some identity issue.

**Pull request merged**